src/shell/shell.zig

```diff
@@ -342,24 +343,293 @@ pub const AST = struct {
 
     pub const Expr = union(Expr.Tag) {
         assign: []Assign,
-        cond: *Conditional,
+        binary: *Binary,
         pipeline: *Pipeline,
         cmd: *Cmd,
         subshell: Script,
+        @"if": *If,
+        condexpr: *CondExpr,
+        /// Valid async (`&`) expressions:
+        /// - pipeline
+        /// - cmd
+        /// - subshell
+        /// - if
+        /// - condexpr
+        /// Note that commands in a pipeline cannot be async
+        /// TODO: Extra indirection for essentially a boolean feels bad for performance
+        /// could probably find a more efficient way to encode this information.
+        @"async": *Expr,
 
         pub fn asPipelineItem(this: *Expr) ?PipelineItem {
             return switch (this.*) {
                 .assign => .{ .assigns = this.assign },
                 .cmd => .{ .cmd = this.cmd },
                 .subshell => .{ .subshell = this.subshell },
+                .@"if" => .{ .@"if" = this.@"if" },
+                .condexpr => .{ .condexpr = this.condexpr },
                 else => null,
             };
         }
 
-        pub const Tag = enum { assign, cond, pipeline, cmd, subshell };
+        pub const Tag = enum {
+            assign,
+            binary,
+            pipeline,
+            cmd,
+            subshell,
+            @"if",
+            condexpr,
+            @"async",
+        };
     };
 
-    pub const Conditional = struct {
+    /// https://www.gnu.org/software/bash/manual/bash.html#Bash-Conditional-Expressions
+    pub const CondExpr = struct {
+        op: Op,
+        args: ArgList = ArgList.zeroes,
+
+        const ArgList = SmolList(Atom, 2);
+
+        // args: SmolList(1, comptime INLINED_MAX: comptime_int)
+        pub const Op = enum {
+            /// -a file
+            ///   True if file exists.
+            @"-a",
```

Comment:
is `-a` normally a flag? it's already covered by `-e`

Reply:
Yep, at first I thought it was gpt hallucinating when it generated this struct, but for some reason -a and -e do the same thing according to the gnu bash manual: https://www.gnu.org/software/bash/manual/bash.html#Bash-Conditional-Expressions

Reply:
oh interesting